### DISCUSSION TITLE
Why API called twice?

### DISCUSSION THREAD

**Discussion by devhyunjae** [TanStack/query#7548] [Q&A] [answered]
demo: https://tz98lg.csb.app/
code: https://codesandbox.io/p/sandbox/react-query-api-twice-call-tz98lg

<img width="1190" alt="Screenshot 2024-06-11 at 11 25 49 AM" src="https://github.com/TanStack/query/assets/12890257/2e0cae93-cbde-4a05-aa16-4ce9b0b1694c">

I know it will work when i pass as a prop. But my real world code has a deeeeeep children components and I don't want to drill down this as a prop and react-query does the magic to fetch the api only once. Can somebody help me? Thanks

**Comment by TkDodo** [comment, marked as answer]
Because the default `staleTime` is zero and you conditionally render components. I'm describing this exact scenario here:

https://tkdodo.eu/blog/react-query-as-a-state-manager

**Reply by devhyunjae** [reply]
Thanks I tried adding staleTime and refetchOnMount: false. delete the conditional render and always render the list.
But still it is calling twice...😰

**Reply by TkDodo** [reply]
you have set the values on an unused queryClient in `App.tsx`. The one that gets passed to the `QueryClientProvider` is in `index.tsx`, and that one still has the default settings.

**Reply by devhyunjae** [reply]
Thanks mate
